test/standalone/child_process/main.zig

```diff
@@ -54,3 +56,29 @@ fn testError(comptime fmt: []const u8, args: anytype) void {
     }
     parent_test_error = true;
 }
+
+fn testSpawnRelative(allocator: std.mem.Allocator, child_path: []const u8) !void {
+    // regression test: ensure we can launch processes "above" the current working directory.
+    const deeper_dir_path = "./1/2/3/4/5";
+    const child_dir_path = std.fs.path.dirname(child_path) orelse return error.BadChildPath;
+    var child_dir = try std.fs.cwd().openDir(child_dir_path, .{});
+    defer child_dir.close();
+
+    try child_dir.setAsCwd();
+    try child_dir.makePath(deeper_dir_path);
+    var deeper_dir = try child_dir.openDir(deeper_dir_path, .{});
+    defer deeper_dir.close();
+    try deeper_dir.setAsCwd();
+    const rel_path = try std.fs.path.relative(allocator, deeper_dir_path, std.fs.path.basename(child_path));
+    defer allocator.free(rel_path);
+
+    const result = try std.process.Child.run(.{ .allocator = allocator, .argv = &.{rel_path} });
+
+    allocator.free(result.stdout);
+    allocator.free(result.stderr);
+
+    const res = std.process.Child.run(.{ .allocator = allocator, .argv = &.{"../../notafile"} });
+
+    // would expect this to be the same error on all platforms. pre-fix windows would return BadPathName instead.
```

Comment:
```suggestion
    // would expect this to be the same error on all platforms.
```

A comment isn't a good place for this info, the part about "pre-fix" behavior can go in the commit message instead